include/swift/AST/DiagnosticsSema.def

```diff
@@ -1404,9 +1404,9 @@ NOTE(decl_multiple_defaulted_closure_parameters,none,
      "%0 contains defaulted closure parameters %1 and %2",
      (DeclName, Identifier, Identifier))
 NOTE(candidate_with_extraneous_args,none,
-      "candidate %0 requires %1 argument%s1, "
-      "but %2 %select{were|was}3 %select{provided|used in closure body}4",
-      (Type, unsigned, unsigned, bool, bool))
+      "contextual type %0 requires %1 argument%s1, "
+      "but %2 %select{were|was}3 %select{provided by candidate of type %5|used in closure body}4",
+      (Type, unsigned, unsigned, bool, bool, Type))
```

Comment:
Huh, apparently this diagnostic isn’t even tested anywhere.

Reply:
Yes, that was why it may had be overlooked. But we are adding at least on case now. Still couldn't figure out a closure repro tho.